src/coreclr/jit/gentree.cpp

```diff
@@ -8258,6 +8258,11 @@ GenTreeLclVar* Compiler::gtNewStoreLclVarNode(unsigned lclNum, GenTree* value)
 
     gtInitializeStoreNode(store, value);
 
+    if (!compImportationDone)
+    {
+        impSetLclVal(lclNum, value);
```

Comment:
There is no guarantee that store nodes are created in execution order, so this cannot be done from here (I also think it shouldn't be from a hygiene perspective). The proper place to do this would be in the importer in a place where we have guarantees about the execution order (e.g. when spilling a relevant store in the importer).